drivers/net/ethernet/microchip/enc28j60.c

```diff
@@ -1595,7 +1578,7 @@ static int enc28j60_probe(struct spi_device *spi)
 	/* Board setup must set the relevant edge trigger type;
 	 * level triggers won't currently work.
 	 */
-	ret = request_irq(spi->irq, enc28j60_irq, 0, DRV_NAME, priv);
+	ret = request_threaded_irq(spi->irq, NULL, enc28j60_irq, IRQF_ONESHOT, DRV_NAME, priv);
```

Comment:
With a threaded irq we cannot programmatically adjust the priority for rx as we can for the tx thread (we still could do this via userspace commands though). If we want to be able to set the priority for tx and rx threads programmatically we have to create or own thread here, too.

Reply:
I think the idea here was just to use threaded irq to replace the "irq + workqueue".